build-support/bin/ci.py

```diff
@@ -417,19 +478,60 @@ def run_jvm_tests() -> None:
   )
 
 
-def run_integration_tests(*, shard: Optional[str]) -> None:
+def run_integration_tests_v1(*, shard: Optional[str]) -> None:
   check_pants_pex_exists()
-  all_targets = get_all_python_tests(tag="+integration")
+  target_sets = TestTargetSets.calculate(
+    test_type=TestType.integration,
+    default_behavior=DefaultTestBehavior.v1_no_chroot,
+    remote_execution_enabled=False
+  )
   command = ["./pants.pex", "test.pytest"]
   if shard is not None:
     command.append(f"--test-pytest-test-shard={shard}")
   with travis_section("IntegrationTests", f"Running Pants Integration tests {shard if shard is not None else ''}"):
     try:
-      subprocess.run(command + sorted(all_targets) + PYTEST_PASSTHRU_ARGS, check=True)
+      subprocess.run(
+        command + ["--test-pytest-chroot"] + sorted(target_sets.v1_chroot) + PYTEST_PASSTHRU_ARGS,
+        check=True
+      )
+      subprocess.run(command + sorted(target_sets.v1_no_chroot) + PYTEST_PASSTHRU_ARGS, check=True)
     except subprocess.CalledProcessError:
       die("Integration test failure.")
 
 
+def run_integration_tests_v2(*, remote_execution_enabled: bool) -> None:
```

Comment:
Should this be a separate method/shard still? Or should we merge this into one shard? We can ask RBE for more quota if we don't have enough parallelism there.

Downside I can think of is that we'll get a lot more output than we probably want for successful runs. But it might be a good time to add a "render success output" option to the generic v2 test runner to skip that anyway.

Reply:
Two questions here:

1) Should `ci.py` have an option `--integration-tests-v2`? I think absolutely yes. This makes it easier to run `ci.py` to test specific things, like test just the integration tests vs unit tests. Notably, this script can have multiple options at a time so we could run something like `ci.py --unit-tests --integration-tests-v2 --remote-execution-enabled`.

2) Should `.travis.yml` have one shard for both remote tasks? For now, I think no.
   * Integration test remoting is still very alpha. Imo it's nice to isolate. For example, maybe we find we need to use `travis_wait` with integration tests but not unit tests.
   * Less confusing to CI users. The shard name "Unit tests and some remoted integration tests" isn't particularly clear to me.

Reply:
> Should ci.py have an option --integration-tests-v2? I think absolutely yes. This makes it easier to run ci.py to test specific things, like test just the integration tests vs unit tests. Notably, this script can have multiple options at a time so we could run something like ci.py --unit-tests --integration-tests-v2 --remote-execution-enabled.

IMO, ci.py should not be necessary to do anything, and actually running "everything that runs in CI" is generally not a good idea. If that looks like being able to emit the `./pants` command that _would_ have run so that someone can whittle it down to just the specific test that they want to run, then that would be my preference vs increasing the complexity of ci.py.

> Less confusing to CI users. The shard name "Unit tests and some remoted integration tests" isn't particularly clear to me.

I buy this from a clarity perspective. But with concurrency in play, the distinction between unit and integration is less relevant, and we might consider prioritizing tests by their timeouts/runtimes in the v2 runner (and starting longer running tests first) instead of actually tagging things integration/unit. Twitter does not use the integration/unit labels internally, and I'm not sure that Pants needs to either.

Reply:
> But with concurrency in play, the distinction between unit and integration is less relevant, and we might consider prioritizing tests by their timeouts/runtimes in the v2 runner (and starting longer running tests first) instead of actually tagging things integration/unit.

I think this might make sense. However, I'd prefer that we save this for a followup change. Our infrastructure is centered around the integration tag right now and we would need to redesign several things to change this. Here, my priority is landing the ability to remote integration tests so that we can start growing the whitelist.

Reply:
Fine with me.